### ISSUE TITLE
Issue with header config for JWT authentication

### ISSUE THREAD

**Issue by josephrace** [feathersjs/generator-feathers#278]
I am currently working through the chat application tutorial but have run into an issue with the configuration generated for JWT authentication.

The authentication generator sets the header config as an object - https://github.com/feathersjs/generator-feathers/blob/master/generators/authentication/index.js#L79

But in the feathers-authentication-jwt module it expects `header` to be a string - https://github.com/feathersjs/feathers-authentication-jwt/blob/master/src/index.js#L39-L41

I'm able to reproduce this each time I try to generate a new app with authentication.

feathers-cli version: 2.3.6

```
feathers generate app

? Project name feathers-test
? Description
? What folder should the source files live in? src
? Which package manager are you using (has to be installed globally)? Yarn
? What type of API are you making? REST, Realtime via Socket.io
```

```
feathers generate authentication

? What authentication providers do you want to use? Other PassportJS strategies not in this list can still be configured manually. Username + Password (Local)
? What is the name of the user (entity) service? users
? What kind of service is it? NeDB
? What is the database connection string? nedb://../data
```

```
yarn start

/Users/joseph/Projects/learning/feathers/feathers-test/node_modules/feathers-authentication-jwt/lib/index.js:58
      throw new Error('You must provide a \'header\' in your authentication configuration or pass one explicitly');
      ^

Error: You must provide a 'header' in your authentication configuration or pass one explicitly
    at Function.jwtAuth (/Users/joseph/Projects/learning/feathers/feathers-test/node_modules/feathers-authentication-jwt/lib/index.js:58:13)
    at Function.configure (/Users/joseph/Projects/learning/feathers/feathers-test/node_modules/feathers/lib/application.js:150:8)
    at Function.module.exports (/Users/joseph/Projects/learning/feathers/feathers-test/src/authentication.js:12:7)
    at Function.configure (/Users/joseph/Projects/learning/feathers/feathers-test/node_modules/feathers/lib/application.js:150:8)
    at Object.<anonymous> (/Users/joseph/Projects/learning/feathers/feathers-test/src/app.js:44:5)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
error Command failed with exit code 1.
```

**Comment by josephrace** [comment]
I notice that someone else appears to be having the same problem and has created an issue in the feathers-authentication-jwt repo instead - https://github.com/feathersjs/feathers-authentication-jwt/issues/34

**Comment by josephrace** [comment]
It seems like this is the commit that has changed things - https://github.com/feathersjs/generator-feathers/commit/680867fb8c012ebe638ed4ec32b90e751835d6ff#diff-eca141b2154850e2c3c63b65110241de

Now that these options are being provided to `jwt` it seems like there might be an issue with how they're merged on https://github.com/feathersjs/feathers-authentication-jwt/blob/master/src/index.js#L37

This is my `jwtSettings` object after merge:

```
{ name: 'jwt',
  bodyKey: 'accessToken',
  secret: '...',
  header: { type: 'access' },
  entity: 'user',
  service: 'users',
  passReqToCallback: true,
  session: false,
  jwt:
   { header: { typ: 'access', type: 'access' },
     audience: 'https://yourdomain.com',
     subject: 'anonymous',
     issuer: 'feathers',
     algorithm: 'HS256',
     expiresIn: '1d' },
  audience: 'https://yourdomain.com',
  subject: 'anonymous',
  issuer: 'feathers',
  algorithm: 'HS256',
  expiresIn: '1d' }
```

**Comment by josephrace** [comment]
Others have reported the same issue elsewhere:

https://github.com/feathersjs/feathers-docs/issues/837
https://github.com/feathersjs/feathers-authentication-jwt/issues/34
https://github.com/feathersjs/feathers-authentication-jwt/issues/35

**Comment by daffl** [comment]
Fixed in `feathers-cli@2.3.7`. The error can be fixed in a generated application by changing

```js
app.configure(jwt(config.jwt));
app.configure(local(config.local));
```

To

```js
app.configure(jwt());
app.configure(local());
```

**Comment by josephrace** [comment]
Thanks!
